sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1379,3 +1379,107 @@ case class GpuStringSplit(str: Expression, regex: Expression, limit: Expression)
       limit: GpuScalar): ColumnVector =
     throw new IllegalStateException("This is not supported yet")
 }
+
+class GpuStringToMapMeta(
+    expr: StringToMap,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+    extends TernaryExprMeta[StringToMap](expr, conf, parent, rule) {
+  import GpuOverrides._
+
+  private def checkRegex(delimExpr: Expression) : Unit = {
+    extractLit(delimExpr).foreach { delim =>
+      val str = delim.value.asInstanceOf[UTF8String]
+      if (str != null){
+        //
+        // Currently, we don't support regex. However, we can't check for regex very accurately.
+        // As such, just treat the delimiter string as a literal string without regex.
+        // In the future, we can re-enable regex check when it is improved.
+        //
+        // if(!canRegexpBeTreatedLikeARegularString(str)) {
+        //   willNotWorkOnGpu("str_to_map does not support regular expression delimiter(s)")
+        // }
+        if (str.numChars() == 0) {
+          willNotWorkOnGpu("delimiter is empty")
+        }
+      } else {
+        willNotWorkOnGpu("delimiter is null")
+      }
+    }
+  }
+
+  override def tagExprForGpu(): Unit = {
+    checkRegex(expr.pairDelim)
+    checkRegex(expr.keyValueDelim)
+  }
+
+  override def convertToGpu(str: Expression,
+                            pairDelim: Expression,
+                            keyValueDelim: Expression): GpuExpression =
+    GpuStringToMap(str, pairDelim, keyValueDelim)
+}
+
+case class GpuStringToMap(strs: Expression, pairDelim: Expression, keyValueDelim: Expression)
+    extends GpuExpression with ExpectsInputTypes with NullIntolerant {
+
+  def this(str: Expression, pairDelim: Expression) = {
+    this(str, pairDelim, GpuLiteral(":", StringType))
+  }
+
+  def this(str: Expression) = {
+    this(str, GpuLiteral(",", StringType), GpuLiteral(":", StringType))
+  }
```

Comment:
Not sure if these constructors are needed, because they are never be called.

Reply:
They are not needed. Some times it is nice to match what Spark does, but here it is just dead code.